Lib/test/test_operator.py

```diff
@@ -531,27 +531,33 @@ def test_concat(self):
         self.assertEqual(data2, 'ab')  # must not change
 
         # Subclasses:
-        class ListSubclass(list): ...
+        class ListSubclass(list):
+            pass
 
         data1 = ListSubclass([1, 2])
         data2 = ListSubclass(['a', 'b'])
-        self.assertEqual(operator.concat(data1, data2),
-                         ListSubclass([1, 2, 'a', 'b']))
-        self.assertEqual(operator.concat(data1, data2), data1 + data2)
+
+        res = operator.concat(data1, data2)
+        self.assertIsInstance(res, list)
+        self.assertEqual(res, [1, 2, 'a', 'b'])
+        self.assertIsInstance(data1, ListSubclass)
         self.assertEqual(data1, ListSubclass([1, 2]))  # must not change
+        self.assertIsInstance(data2, ListSubclass)
         self.assertEqual(data2, ListSubclass(['a', 'b']))  # must not change
 
         # Custom type with `__add__`:
         class TupleSubclass(tuple):
             def __add__(self, other):
-                return other + self
+                return TupleSubclass(other + self)
 
         data1 = TupleSubclass([1, 2])
         data2 = ('a', 'b')
-        self.assertEqual(operator.concat(data1, data2),
-                         TupleSubclass(['a', 'b', 1, 2]))
-        self.assertEqual(operator.concat(data1, data2), data1 + data2)
+        res = operator.concat(data1, data2)
+        self.assertIsInstance(res, TupleSubclass)
+        self.assertEqual(res, TupleSubclass(['a', 'b', 1, 2]))
+        self.assertIsInstance(data1, TupleSubclass)
```

Comment:
But this is still questionable. I can't think of a scenario where this would fail.

Or do you have coverage results showing this is needed? (Where?)

Ditto for data2 and again in the following block of tests.